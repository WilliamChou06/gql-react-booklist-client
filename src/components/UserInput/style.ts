import styled from 'styled-components';

export const UserInputWrapper = styled.div`
  flex: 1;
  padding: 12px;
`