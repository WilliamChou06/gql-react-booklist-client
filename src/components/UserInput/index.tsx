import React, { lazy, Suspense } from 'react';
import { UserInputWrapper } from './style';

// Lazy loading
const AddAuthor = lazy(() => import('../AddAuthor'))
const AddBook = lazy(() => import('../AddBook'))


const UserInput = () => {
  return (
    <UserInputWrapper >
      <Suspense fallback={<div>Loading...</div>}>
        <AddAuthor />
      </Suspense>
      <br></br>
      <hr></hr>
      <h1>HELLOO</h1>
      <Suspense fallback={<div>Loading...</div>}>
        <AddBook />
      </Suspense>
      </UserInputWrapper>
      );
    };
    
export default UserInput;