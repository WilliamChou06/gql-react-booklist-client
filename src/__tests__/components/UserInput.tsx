import React from 'react'
import { shallow } from 'enzyme'

import UserInput from '../../components/UserInput';

it('should render UserInput component correctly', () => {
  const wrapper = shallow(<UserInput />);

  expect(wrapper).toMatchSnapshot();
});