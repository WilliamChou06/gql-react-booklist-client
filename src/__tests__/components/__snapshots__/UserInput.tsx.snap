// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render UserInput component correctly 1`] = `ShallowWrapper {}`;
